src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.h

```diff
@@ -343,7 +351,8 @@ using addr_type = uint64_t;
       static std::ofstream mDebugLogStream;
       static bool mFirstBinary;
       unsigned int binaryCounter;
-      unix_socket* sock;
+
+      std::shared_ptr<unix_socket> sock;
```

Comment:
I can't really tell, but is this really shared ownership?  If it is exclusive then use std::unique_ptr.

Reply:
Hoho! Removing the macros???
In the meantime, just having the macro using it by const reference

Reply:
It is for future clean up CR but not as part of current CR.